### PR TITLE
[video] Versions/Extras Manage Dialogs: Selection improvements

### PR DESCRIPTION
**Pull request by ksooo**
Couple of fixes and improvements for the Video versions/extras manage dialogs:

1. Fix initial selection to be set to default video version / first extra
2. Update button state after "Rename"
3. Highlight default video version in list of available versions.
screenshot00005

5. Update buttons whenever the selection in list changes (e.g. up/down key pressed); no longer need to click to update the buttons.

Runtime-tested on macOS, latest Kodi master.

@enen92 this time not that much code to review. Thanks for all your support. ;-)

@HitcherUK @jjd-uk @CrystalP everyone, fyi. Runtime-test feedback appreciated.

### CODE REVIEW

File: xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp
Comment (enen92) [inline xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp:106]: You can probably assign this to a variable and pass it to select. Or use the condition directly in select.
Up to you, readibility might decrease but you kill the if/else branches

**Review by enen92 — APPROVED** [review]
All good with the code. Glad to see one I actually understand :) it's a sign this is going on the right way

**Comment by HitcherUK** [comment]
> 4\. Update buttons whenever the selection in list changes (e.g. up/down key pressed); no longer need to click to update the buttons.

This is so much more intuitive, thanks.

File: xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp
Comment (ksooo) [reply xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp:106]: Yep, will change that.

File: xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp
Comment (CrystalP) [inline xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp:162]: typo, UpdateDefaultVideoVersionSelection()

**Comment by CrystalP** [comment]
That's a step in the right direction, hopefully with skinning there is a way for a highlight to remain on the selected item of the list after going right to select one of the buttons? Could be just me, but the slightly brighter text is barely different from the other lines.

Small problem when opening the screen for the very first time and the first version in the list is not the default.
The highlight is on the first line (a non-default version), but the buttons "Remove" and "Set Default" are not enabled (as if the first line was the default version)
After moving down/up, the buttons are updated as expected

Very minor glitch, only seen because I was scrutinizing the screen: when dismissing the dialog you can see during the closing animation that the "Add version" button disappears, causing the buttons under it to "move up".

Are you going to update the play select screen next to display the default version?

File: xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp
Comment (ksooo) [reply xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp:162]: Haha, thanks.

**Comment by ksooo** [comment]
> there is a way for a highlight to remain on the selected item of the list after going right to select one of the buttons?

Sorry, I do not understand the question? Could you please explain the problem with some more words? Best is step-by-step instruction for reproduction.

> Small problem when opening the screen for the very first time and the first version in the list is not the default.
The highlight is on the first line (a non-default version), but the buttons "Remove" and "Set Default" are not enabled

I can reproduce this. That's clearly a bug I need to fix before we merge this.

> Very minor glitch, only seen because I was scrutinizing the screen: when dismissing the dialog you can see during the closing animation that the "Add version" button disappears, causing the buttons under it to "move up".

Nothing I have the skills for a fix. 

> Are you going to update the play select screen next to display the default version?

I could have a look, yes. What exactly di you mean with "display"? You mean highlight the version like now done in manage dialog?

File: xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp
Comment (ksooo) [reply xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp]: Fixed with latest force-push.

**Comment by ksooo** [comment]
>> Small problem when opening the screen for the very first time and the first version in the list is not the default.
The highlight is on the first line (a non-default version), but the buttons "Remove" and "Set Default" are not enabled

> I can reproduce this. That's clearly a bug I need to fix before we merge this.

Should be fixed with latest force-push.

**Comment by Hitcher** [comment]
> That's a step in the right direction, hopefully with skinning there is a way for a highlight to remain on the selected item of the list after going right to select one of the buttons? Could be just me, but the slightly brighter text is barely different from the other lines.

This is just how it is across the rest of the skin when moving away from the list.

eg 
screenshot00011

screenshot00012

**Comment by Hitcher** [comment]
> Very minor glitch, only seen because I was scrutinizing the screen: when dismissing the dialog you can see during the closing animation that the "Add version" button disappears, causing the buttons under it to "move up".

This is purely because those buttons have visible conditions based on the visible window.

`Window.IsActive(managevideoversions)` and `Window.IsActive(managevideoextras)`

so when the window gets closed they're not true anymore.

**Comment by ksooo** [comment]
> This is just how it is across the rest of the skin when moving away from the list.

Now I understood the question from @CrystalP. According to this answer I take this as no problem, but a feature, right, @Hitcher ?

**Comment by Hitcher** [comment]
> Now I understood the question from @CrystalP. According to this answer I take this as no problem, but a feature, right, @Hitcher ?

Yes, changing it here would mean it should be changed across the skin.

**Comment by ksooo** [comment]
> This is purely because those buttons have visible conditions based on the visible window.

Would it help to let the core control the visible state of the buttons?

**Comment by ksooo** [comment]
Jenkins build this please

**Comment by Hitcher** [comment]
> Would it help to let the core control the visible state of the buttons?

See my edit: `Window.IsVisible` although having their visibility defined in core would also work.

**Comment by ksooo** [comment]
> See my edit: Window.IsVisible will fix it.

Cool. @Hitcher could you open a PR for the fix, please? I don't want to add it here, because not related to selection issues directly.

**Comment by ksooo** [comment]
@CrystalP could you retest to ensure latest force-push has fixed the "Small problem when opening the screen for the very first time and the first version in the list is not the default." and has not introduced any regressions? If you find nothing, this PR should be ready to go in.

**Comment by CrystalP** [comment]
OK, now the screen opens with the default version highlighted, that's even better! However found an inconsistency.
- With a movie that has 2 versions, the default is the 2nd one listed in Manage screen.
- Screen opens with line 2 highlighted + selection: OK
- Press right, "Rename" the version
- The version was renamed, the button is still selected
- Press Left. The 1st version is highlighted.
- Press Right. Focus goes back to Rename button
- Rename the version.
- The default version is renamed, instead of the 1st version that was last highlighted before Rename.

Regarding the visibility of the selected version, it was already problematic before your changes and it's not the focus of this PR so separate PR I guess.
To use the color to indicate the selection for consistency, then something else needs to identify the default version. For example a star bitmap, like this:
image
(first line is colored = selection that the button will apply to, line 2 with the star = default version)


Alternatively I had the settings screen in mind as an example where the category keeps a darker highlight after moving to an individual setting on the right:
image


> > Are you going to update the play select screen next to display the default version?
> 
> I could have a look, yes. What exactly di you mean with "display"? You mean highlight the version like now done in manage dialog?

I meant the "Choose version" dialog when playing from library. Yes highlight / pre-select the default version.

**Comment by ksooo** [comment]
> Regarding the visibility of the selected version, it was already problematic before your changes and it's not the focus of this PR so separate PR I guess.

Right, definitely nothing for this PR. This is a problem throughout the whole skin and should be "fixed" consistently everywhere. Nothing I will/can fix here.

**Comment by ksooo** [comment]
> However found an inconsistency.

Thanks. I will have a look and fix it in this PR before merge.

**Comment by Hitcher** [comment]
> Alternatively I had the settings screen in mind as an example where the category keeps a darker highlight after moving to an individual setting on the right. It's very clear in that screenshot that the displayed settings belong to the "Videos" category;

Yeah, I agree with this and it should be the same for lists in dialogs across the skin.

If I get time I'll see if I can sort out a PR.

**Comment by ksooo** [comment]
> However found an inconsistency.

Fixed with latest push. I will merge the PR once Jenkins is green.

**Pull request merged**
